src/main/java/hudson/remoting/Engine.java

```diff
@@ -403,6 +463,101 @@ public static Engine current() {
 
     private static final Logger LOGGER = Logger.getLogger(Engine.class.getName());
 
+    private static KeyStore getCacertsKeyStore()
+            throws PrivilegedActionException, KeyStoreException, NoSuchProviderException, CertificateException,
+            NoSuchAlgorithmException, IOException {
+        Map<String, String> properties = AccessController.doPrivileged(
+                new PrivilegedExceptionAction<Map<String, String>>() {
+                    public Map<String, String> run() throws Exception {
+                        Map<String, String> result = new HashMap<String, String>();
+                        result.put("trustStore", System.getProperty("javax.net.ssl.trustStore"));
+                        result.put("javaHome", System.getProperty("java.home"));
+                        result.put("trustStoreType",
+                                System.getProperty("javax.net.ssl.trustStoreType", KeyStore.getDefaultType()));
+                        result.put("trustStoreProvider", System.getProperty("javax.net.ssl.trustStoreProvider", ""));
+                        result.put("trustStorePasswd", System.getProperty("javax.net.ssl.trustStorePassword", ""));
+                        return result;
+                    }
+                });
+        KeyStore keystore = null;
+
+        FileInputStream trustStoreStream = null;
+        try {
+            String trustStore = properties.get("trustStore");
+            if (!"NONE".equals(trustStore)) {
+                File trustStoreFile;
+                if (trustStore != null) {
+                    trustStoreFile = new File(trustStore);
+                    trustStoreStream = getFileInputStream(trustStoreFile);
+                } else {
+                    String javaHome = properties.get("javaHome");
+                    trustStoreFile = new File(
+                            javaHome + File.separator + "lib" + File.separator + "security" + File.separator
+                                    + "jssecacerts");
+                    if ((trustStoreStream = getFileInputStream(trustStoreFile)) == null) {
```

Comment:
From what I see in Java documentation, jssecacerts and cacerts have no priority specifications. BTW here we ignore "cacerts" if we can get something from jssecacerts. Is it correct?